src/Refactoring-Core/RBRenameArgumentOrTemporaryRefactoring.class.st

```diff
@@ -7,7 +7,7 @@ The variable declaration an all references in this method are renamed.
 My precondition verify that the new name is a valid variable name and not an existing instance or a class variable name
 "
 Class {
-	#name : #RBRenameTemporaryRefactoring,
+	#name : #RBRenameArgumentOrTemporaryRefactoring,
```

Comment:
I still feel that this new distinguish between temp and arg is wrong. And I believe this complex names prove that. 
What is wrong and where it is not convenient to have just temps variables? 
Arguments are also temp vars. But not all temp vars are arguments. Why this sentence is bad?

Reply:
I'm gonna try to make an argument, we'll see if that makes sense :p
Also, this argument is to the best of my knowledge, so if you find holes in them, please  help me fill those gaps =)

For me, a lot of tools are working on the AST (such as this one).
Therefore, when I interact with a tool, I have the AST's semantics in mind;
The AST differentiate temporaries and arguments (which I personally like).

In the compiler, you have a compiler view, which does not differentiate temporaries and args.
When interacting with the AST, it changes view, it should ask #isArgOrTemp to have the same behavior as on its nodes for #isTemp.
When you know one view, and someone else's comes into play without warning, what do you do? 
You make semantics errors.

As I said, in the context of the sourceCodeRefactor, you have currently 3 users of #isTempNodeSelected.
"1 of them is actually using #isTempOrArgNodeSelected.
2 of them actually want the right semantic that exists for #isTempNodeSelected."
What I meant by that is that semantically, there is 1 that is appliable on both temps and args, and two only on temps.
Meaning that without differentiation, there is 2 extra refactors offered for argument nodes that cannot be used.
I think it's a semantic mistake. 

>Arguments are also temp vars. But not all temp vars are arguments. Why this sentence is bad?
(I think that) if they had a common super class, that showed that, I would agree with this.
And I don't mean VariableNodes because instanceVariables are definitely not temps.

Maybe a simple differentiation between something like #isTemp on one side, and #isTemporaryNode/#isArgumentNode on the other would be enough ? 
I think it's error prone, but this would at least separate the different concepts, the different views.

Let me know what you think !